src/com/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1264,3 +1243,21 @@
   [action-for-unsupported-fn]
   (fail-on-unsupported action-for-unsupported-fn)
   (warn-on-db-deprecation))
+
+(def ^:dynamic *orphaned-path-gc-limit* 200)
+(def ^:dynamic *orphaned-value-gc-limit* 200)
+
+(defn garbage-collect!
+  "Delete any lingering, unassociated data in the database"
+  [db]
+  (time!
+   (:gc metrics)
+   (jdbc/with-transacted-connection db
+     (sql/transaction (delete-unassociated-params!))
+     (sql/transaction (delete-unassociated-environments!))
+     (sql/transaction (delete-unassociated-statuses!)))
+   ;; These require serializable because they make the decision to
+   ;; delete based on row counts in another table.
+   (jdbc/with-transacted-connection' db :serializable
```

Comment:
@senior I've separated the new orphaned facts and paths gc transactions as suggested, but I left the delete-unassociated-* calls together since that's how they were before.